Source/WebInspectorUI/UserInterface/Views/SpreadsheetRulesStyleDetailsPanel.js

```diff
@@ -300,10 +302,29 @@ WI.SpreadsheetRulesStyleDetailsPanel = class SpreadsheetRulesStyleDetailsPanel e
                     pseudoElement = this.nodeStyles.node.beforePseudoElement();
                 else if (pseudoId === WI.CSSManager.PseudoSelectorNames.After)
                     pseudoElement = this.nodeStyles.node.afterPseudoElement();
+
+                let orderedPseudoStyles = WI.DOMNodeStyles.uniqueOrderedStyles(pseudoElementInfo.orderedStyles);
+
+                // Check if there are any styles to show before adding the header.
+                let hasStylesToShow = false;
+                for (let style of orderedPseudoStyles) {
+                    if (this._shouldIncludeStyle(style)) {
+                        hasStylesToShow = true;
+                        break;
+                    }
+                }
+
+                if (!hasStylesToShow)
+                    continue;
+
                 addHeader(WI.UIString("Pseudo-Element"), pseudoElement || pseudoId);
 
-                for (let style of WI.DOMNodeStyles.uniqueOrderedStyles(pseudoElementInfo.orderedStyles))
+                for (let style of orderedPseudoStyles) {
+                    if (!this._shouldIncludeStyle(style))
```

Comment:
having to check this twice is i think not as easy to read as when it was all in a single array (even though it did require an extra allocation)

as such, id go back to what you had before
```js
let orderedPseudoStyles = WI.DOMNodeStyles.uniqueOrderedStyles(pseudoElementInfo.orderedStyles)
    .filter((style) => this._shouldIncludeStyle(style));
```

apologies for leading you down the wrong path